### ISSUE TITLE
Compositelist single selection w/ scan mode sets no aria-label for checkbox column

### ISSUE THREAD

**Issue by joshualamusga1** [microsoft/fluentui#20411]
### Environment Information
- **Package version(s)**: Fluent UI 8.7.36 (demo site)
- **Browser and OS versions**: Edge Chromium on Windows 10 64-bit Enterprise

### Describe the issue:
When a detailslist is in single-select mode, the aria label for the checkbox column header is not set, and there is nothing for screen readers to say except "column 1". It should have a label for screen readers that mentions what the column is for (even though there is no select all checkbox) because (1) the column is still relevant and (2) screen readers will announce the first column after as "column 2" which will probably cause people using screen readers to see what column 1 was.

### Repro:
https://codepen.io/joshualamusga1/pen/YzxxeqN
This codepen is the same as the Detailslist page in the FluentUI demo site, except single select has been set.

1. Open Narrator or NVDA
2. Navigate to the given Codepen and tab to the Name column
3. Switch to scan/read mode if not already activated (usually CapsLock+Space)
4. Press the left arrow key to move to column 1

#### Actual behavior:
Narrator will read out "Column header. Column 1 of 6."

#### Expected behavior:
Narrator should read out "Column header. {column name}. Column 1 of 6.

### Documentation describing expected behavior:
https://dequeuniversity.com/rules/axe/4.3/empty-table-header?application=AxeChrome

### Notes:
It looks like there is an aria-labelledby attribute placed on the checkbox, which for me points to an element called header7-checkTooltip. This element doesn't exist. Probably, that's the source of the issue. Manually changing this with devtools to an aria-label and removing aria-labelledby in this case causes it to read out like "Column header. Column header {name}. Column 1 of 6."

(I won't be able to submit a PR for this.)

**Comment by gouttierre** [comment]
@joshualamusga1 -Thanks for filing this issue with us. To set expectations, the developers will review this issue once capacity allows. However, if you get the time to submit a PR for the fix in the interim, please do so and we will gladly review it.

@layershifter - Would you be able to confirm if this is a regression, or if this behavior is an issue? 
cc @smhigley
